package.json

```diff
@@ -51,9 +51,10 @@
   "peerDependencies": {
     "react": "^16.6.1 || ^17.0.0 || ^18.0.0",
     "react-dom": "^16.6.1 || ^17.0.0 || ^18.0.0",
-    "styled-components": "^5.1.0"
+    "styled-components": "^5.1.0 || ^6.0.0"
```

Comment:
Supporting both seems like it might be risky with significant breaking changes specifically because of how v6 handles nesting as compared to v5.  You can find that documentation [here](https://github.com/styled-components/styled-components/blob/22e8b7f233e12500a68be4268b1d79c5d7f2a661/packages/styled-components/src/types.ts#L152).

Reply:
Hey @Tbhesswebber thanks for commenting. Could you elaborate on your concerns? Definitely want to hear them out as we're considering adding support for v6 as well.

With these changes, we are taking care of shouldForwardProp (mentioned in your link above) on all styled instances within grommet components (see styles.js under files changed). This will ensure proper styling if someone is using styled components v6.

Reply:
I updated the link above (I was hasty with copy pasta, it seems).

```ts
const Foo = styled.button`
  :hover {
    background: black;
  }
`;
```

The above snippet outputs different CSS in v5 and v6. In v5, there's an implicit `&` prepended to `:hover`, so it's applied directly to the element. In v6, it's prepended with `& ` to align with the nesting spec for CSS, so the hover state applies to children of the button rather than the button itself.

Reply:
I went through and added `&` to the places in Grommet where the `&` was being implicitly added by styled components v5 so that we will get the same behavior in v5 and v6. (See the changes in StyledCheckBox.js for an example)

Reply:
Sorry for not being clear, my concern is moreso in the realm of consumers having this syntax in the `.extends` properties in their configuration or in `styled(SomeGrommetComponent)` declarations. Breaking changes that likely won't be documented in Grommet's changelog because they aren't actually changes to Grommet

Reply:
I think we can add a note to the Grommet release where we add styled components v6 support, so that this doesn't catch anyone by surprise. We have a similar message on the alpha release we did for styled components v6 support: https://github.com/grommet/grommet/releases/tag/v2.35.0-alpha.1.
But ultimately since the breaking change is not caused by Grommet and is a result of the styled-components upgrade then I think the person upgrading would just have to refer to the styled-components upgrade documentation which calls out the breaking changes.
Since we will have support for both v5 and v6 of styled components it is up to the consumer if they want to upgrade to v6 or not.